src/utils/aggregateEdges.ts

```diff
@@ -0,0 +1,75 @@
+import { InternalGraphEdge } from '../types';
+
+/**
+ * Groups edges by their source and target nodes.
+ * @param edges Array of edges to group
+ * @returns Map with source-target pairs as keys and arrays of edges as values
+ */
+export const groupEdgesBySourceTarget = (
+  edges: InternalGraphEdge[]
+): Map<string, InternalGraphEdge[]> => {
+  const edgeGroups = new Map<string, InternalGraphEdge[]>();
+
+  // Filter out undefined edges
+  const validEdges = edges.filter(edge => edge && edge.source && edge.target);
```

Comment:
I'm wondering if there might be a better way to avoid the looping so much. Graphology has some powerful functions that might give you what you need natively in a more optimized way. https://graphology.github.io/read.html#edge